src/components/Footer/FooterNav/FooterNav.tsx

```diff
@@ -1,36 +1,95 @@
 import React from 'react'
 import classnames from 'classnames'
 
+type ExtendedNavLinks = [React.ReactNode[]]
+
 type FooterNavProps = {
   big?: boolean
   medium?: boolean
   slim?: boolean
-  links: React.ReactNode[]
+  /* 
+    Union type. Array of navigation links or multidimensional array of ExtendedNavLinks.
+    ExtendedNavLinks are ordered sub arrays that will be displayed as columns, with the first element used as the section heading.
+    ExtendedNavLinks can only be used with "big" prop size
+  */
+  links: React.ReactNode[] | ExtendedNavLinks
+}
+
+function isExtendedNavLinks(
+  links: React.ReactNode[] | ExtendedNavLinks
+): links is ExtendedNavLinks {
+  return (links as ExtendedNavLinks)[0].constructor === Array
 }
 
 export const FooterNav = (
   props: FooterNavProps & React.HTMLAttributes<HTMLElement>
 ): React.ReactElement => {
-  const { medium, slim, links, ...elementAttributes } = props
+  const { big, medium, slim, links, ...elementAttributes } = props
 
-  const navClasses = classnames(`usa-footer__nav`, elementAttributes.className)
   const listItemClasses = classnames(
     'desktop:grid-col-auto usa-footer__primary-content',
     {
-      'mobile-lg:grid-col-4': medium,
+      'mobile-lg:grid-col-4': big || medium,
       'mobile-lg:grid-col-6': slim,
     }
   )
 
   return (
-    <nav className={navClasses} aria-label="Footer navigation">
-      <ul className="grid-row grid-gap">
-        {links.map((link, i) => (
-          <li key={`navLink-${i}`} className={listItemClasses}>
+    <nav
+      {...elementAttributes}
+      className="usa-footer__nav"
+      aria-label="Footer navigation">
+      {big && isExtendedNavLinks(links) && <ExtendedNav nestedLinks={links} />}
+
+      {!isExtendedNavLinks(links) && (
+        <ul className="grid-row grid-gap">
```

Comment:
Just as a heads up, I created a list component that could be used here too. It's not merged in yet though, so 🤷 
https://github.com/trussworks/react-uswds/blob/61535c9510d4d528f02b09aaaa45de8ff0ebd789/src/components/header/List/List.stories.tsx